lib/models/storage-event.js

```diff
@@ -3,43 +3,99 @@
 const mongoose = require('mongoose');
 const utils = require('../utils');
 /**
- * Represents the history of storage related summary 
+ * Represents the history of storage related summary
  * statistics for buckets and their associated files
  * @constructor
  */
 
 var StorageEvent = new mongoose.Schema({
-  bucket: {
-    type: mongoose.Schema.Types.ObjectId,
-    ref: 'Bucket',
-    required: true
-  },
-  bucketEntry: {
-    type: mongoose.Schema.Types.ObjectId,
-    ref: 'BucketEntry',
-    required: true
+  token: {
+    type: String,
+    required: false,
+    unique: true
```

Comment:
True that 160bit is not as much as 256bits, however it's still huge. 160bit hashes are used for pay-to-public-key hash for Bitcoin transactions, and issues have been raised before and addressed with segwit updates, but the attack is still very impractical.

Reply:
Ethereum nodeIDs are also 160bits

Reply:
Bitcoin has only 900K unique addresses. The same address can be generated twice.

The storj network is different. Every farmer will "brute force token generation" 24/7. My farmer is generating 50K tokens per month. Lets say we have only 10K active farmer. 500M brute force tokens per month. Even if the chance is very low we have to deal with so many tokens that I would expect a few collsions.

Reply:
It would take roughly 243,583,606,200,000,000,000,000,000,000,000,000,000 years to have collision at that rate. There are [2 ^ 160](https://www.wolframalpha.com/input/?i=2+%5E+160) possible tokens with 160 bits. Furthermore there isn't a restriction on length, and it could be 32 bytes instead of 20 bytes (another topic).